### PR TITLE
Fix panic when response is nil in DownloadRepoFromFile

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/froggit-go/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `go fmt ./...` for formatting the code before submitting the pull request.
- [x] This feature is included on all supported VCS providers - GitHub, Bitbucket cloud, Bitbucket server, and GitLab.

---

### CODE REVIEW

File: vcsclient/github.go [hunk 1]
```diff
@@ -573,24 +574,21 @@ func (client *GitHubClient) DownloadFileFromRepo(ctx context.Context, owner, rep
...
-			}
+			err = errors.Join(err, body.Close())
 		}
 	}()
-	if response != nil && response.StatusCode != http.StatusOK {
-		return nil, response.StatusCode, fmt.Errorf("expected %d status code while received %d status code with error:\n%s", http.StatusOK, response.StatusCode, err)
+	if response != nil && response.Response != nil {
+		statusCode = response.StatusCode
+	}
+	if response != nil && response.Response != nil && response.Response.StatusCode != http.StatusOK {
```
Comment (sverdlov93) [inline vcsclient/github.go:583]:
```suggestion
	if statusCode != http.StatusOK {
```

File: vcsclient/github.go [hunk 2]
```diff
@@ -573,24 +574,21 @@ func (client *GitHubClient) DownloadFileFromRepo(ctx context.Context, owner, rep
...
+		err = fmt.Errorf("expected %d status code while received %d status code with error:\n%s", http.StatusOK, response.StatusCode, err)
```
Comment (sverdlov93) [inline vcsclient/github.go:584]: how said there is err?

File: vcsclient/gitlab.go [hunk 3]
```diff
@@ -447,19 +447,19 @@ func (client *GitLabClient) GetRepositoryEnvironmentInfo(_ context.Context, _, _
...
-	if response != nil && response.StatusCode != http.StatusOK {
-		return nil, response.StatusCode, fmt.Errorf("expected %d status code while received %d status code with error:\n%s", http.StatusOK, response.StatusCode, err)
+	var statusCode int
+	if response != nil && response.Response != nil {
+		statusCode = response.Response.StatusCode
 	}
-	if err != nil {
-		return nil, 0, err
+	if statusCode != http.StatusOK {
+		return nil, statusCode, fmt.Errorf("expected %d status code while received %d status code with error:\n%s", http.StatusOK, response.StatusCode, err)
```
Comment (sverdlov93) [inline vcsclient/gitlab.go:455]: who said there is error?

File: vcsclient/gitlab.go [hunk 4]
```diff
@@ -447,19 +447,19 @@ func (client *GitLabClient) GetRepositoryEnvironmentInfo(_ context.Context, _, _
...
-
-	content, err := base64.StdEncoding.DecodeString(file.Content)
 	if err != nil {
-		return nil, response.StatusCode, err
+		return nil, statusCode, err
 	}
 
-	return content, response.StatusCode, err
+	content, err := base64.StdEncoding.DecodeString(file.Content)
+	return content, response.Response.StatusCode, err
```
Comment (sverdlov93) [inline vcsclient/gitlab.go:462]: who said response is not nil? how said response.response is not nil?

File: vcsclient/github.go [hunk 5]
```diff
@@ -573,24 +574,21 @@ func (client *GitHubClient) DownloadFileFromRepo(ctx context.Context, owner, rep
 	body, response, err := ghClient.Repositories.DownloadContents(ctx, owner, repository, path, &github.RepositoryContentGetOptions{Ref: branch})
```
Comment (sverdlov93) [inline vcsclient/github.go:574]: let's change it to bitbacketResponse / githubResponse / gitlabResponse for all following changes:

File: vcsclient/github.go (hunk 1 above)
Comment (omerzi) [reply vcsclient/github.go:583]: done

File: vcsclient/gitlab.go (hunk 3 above)
Comment (omerzi) [reply vcsclient/gitlab.go:455]: done

File: vcsclient/gitlab.go (hunk 4 above)
Comment (omerzi) [reply vcsclient/gitlab.go:462]: done

File: vcsclient/github.go (hunk 2 above)
Comment (omerzi) [reply vcsclient/github.go:584]: done

**Pull request merged**